compiler/resolution/lifetime.cpp

```diff
@@ -1220,6 +1631,19 @@ static bool containsOwnedClass(Type* type) {
   return false;
 }
 
+static bool isOrContainsBorrowedClass(Type* type) {
+  type = type->getValType();
+
+  if (AggregateType* at = toAggregateType(type)) {
+    if (at->isClass())
```

Comment:
If there is a function that tests specifically for a "borrow" type, it should probably be used here instead.

Reply:
there isn't a global function for that right now, but I cleaned up this area of the checker implementation in other ways.